autosubmit/job/job_list.py

```diff
@@ -2264,10 +2264,9 @@ def get_jobs_by_section(self, section_list: list, banned_jobs: list = None,
                     jobs.append(job)
         return jobs
 
-    def get_in_queue_grouped_id(self, platform):
-        # type: (object) -> Dict[int, List[Job]]
-        jobs = self.get_in_queue(platform)
-        jobs_by_id = dict()
+    def get_in_queue_grouped_id(self, platform) -> dict[str, list[Job]]:
+        jobs: list[Job] = self.get_in_queue(platform)
+        jobs_by_id: dict[str, list[Job]] = dict()
```

Comment:
Adding types, as they show in the IDE where the error is. No other changes here yet.